pkg/portal/portal.go

```diff
@@ -132,43 +129,40 @@ func NewPortal(env env.Core,
 	}
 }
 
-func (p *portal) setupRouter() error {
-	if p.baseRouter != nil {
-		return fmt.Errorf("can't setup twice")
-	}
-
+func (p *portal) setupRouter(kconfig *kubeconfig.Kubeconfig, prom *prometheus.Prometheus, sshStruct *ssh.SSH) (*mux.Router, error) {
 	r := mux.NewRouter()
 	r.Use(middleware.Panic(p.log))
 
 	assetv1, err := assets.EmbeddedFiles.ReadFile("v1/build/index.html")
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	assetv2, err := assets.EmbeddedFiles.ReadFile("v2/build/index.html")
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	p.templateV1, err = template.New("index.html").Parse(string(assetv1))
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	p.templateV2, err = template.New("index.html").Parse(string(assetv2))
 	if err != nil {
-		return err
+		return nil, err
 	}
```

Comment:
Have you considered moving this logic to another small function?

Reply:
yes, but I want to stay as close as possible to the structure that we had before, we can use smaller functions later.
here I want the scope of the PRs to be as small as possible because it is a very complex piece of code and it instrumental to our operations

Reply:
oh, got you, really makes sense!